.github/workflows/publish.yml

```
@@ -10,23 +10,23 @@ jobs:
 
     steps:
       - name: Checkout project
-        uses: actions/checkout@v2
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
         with:
           ref: main
           fetch-depth: 0
 
       - name: Cache Maven
-        uses: actions/cache@v2.1.4
+        uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
```

Looks like  the [latest major version](https://github.com/actions/cache#restoring-and-saving-cache-using-a-single-action) to use for cache is v3.

```suggestion
        uses: actions/cache@v3
```

---

yes, it's 3.3.2, it's safer to use teh commit hash and a comment, dependabot knows what to do going forward